### DISCUSSION TITLE
Testing with Ventura (MacOS 13) preview

### DISCUSSION THREAD

**Discussion by RhetTbull** [RhetTbull/osxphotos#711] [General]
Any osxphotos users have access to the macOS Ventura developer preview? I'd love to get an early start on upgrading osxphotos to work with Ventura but need a Ventura Photos Library.

**Comment by RhetTbull** [comment]
Thanks to a user on Reddit's [r/MacOSBeta](https://www.reddit.com/r/MacOSBeta/) I was able to get a Ventura test library.  So far the [changes](https://github.com/RhetTbull/osxphotos/wiki/Photos-7-to-Photos-8-diff) appear to be relatively minor.  I won't be able to test some of the new Ventura features such as locked hidden photos and the new iCloud family sharing until I get access to a Ventura machine but I think osxphotos will be able to run fine on Ventura.  Of course, it's possible Apple has implemented new SIP protections that would prevent raw access to the library...can't test those until Ventura is more widely available and I can get user feedback.

**Reply by Rott-Apple** [reply]
Hi Rhet. Any updates on osxphotos support for iCloud Shared Photo Library in macOS Ventura? We know that the shared library's creator provides iCloud storage for everyone, but everyone has equal write/write rights to it. Will there be a difference in how the creator and the other users export the shared photos using osxphotos?

**Reply by RhetTbull** [reply]
Hi @Rott-Apple I don't yet know how osxphotos will handle this. My Mac is an old machine not capable of running Ventura so until I upgrade the Mac or am somehow able to get access to a Ventura machine I can't do any specific testing or code changes.  If any osxphotos users do have Ventura and can test against a library with iCloud Shared Photo Library, I might be able to get enough data to work with.

**Comment by RhetTbull** [comment]
```
pytest tests/test_ventura_dev_preview_13_0_0.py
==================================================================== test session starts =====================================================================
platform darwin -- Python 3.8.12, pytest-7.0.1, pluggy-1.0.0
rootdir: /Users/rhet/Dropbox/Code/osxphotos, configfile: pytest.ini
plugins: mock-3.7.0
collected 77 items

tests/test_ventura_dev_preview_13_0_0.py .............................................................................                                 [100%]

===================================================================== 77 passed in 3.20s =====================================================================
```
